manifests/init.pp

```diff
@@ -176,6 +194,12 @@
     $server_user_real = $server_user
   }
 
+  if $service_provider == 'USE_DEFAULTS' {
+    $service_provider_real = $default_service_provider
+  } else {
+    $service_provider_real = $service_provider
+  }
+
```

Comment:
please add validation, so that if service_provider != undef you validate it with is_string()